pcdsdevices/targets.py

```diff
@@ -260,3 +264,122 @@ def move(self, nxspaces, nyspaces, wait=False):
 
         self.x.mv(xpos, wait=wait)
         self.y.mv(ypos, wait=wait)
+
+    def tweak(self):
+        """
+        Call the tweak function from `pcdsdevice.interface`.
+
+        Use the Left arrow to move x motor left.
+        Use the Right arrow to move x motor right.
+        Use the Down arrow to move y motor down.
+        Use the Up arrow to move y motor up.
+        Use Shift & Up arrow to scale*2.
+        Use Shift & Down arrow to scale/2.
+        Press q to quit.
+        """
+        tweak_base(self.x, self.y)
+
+    def set_presets(self):
+        """
+        Save three preset coordinate points.
+        """
+        print('Setting coordinates for (N, 0) bottom left corner')
+        self.tweak()
+        pos = [self.x.position, self.y.position]
+        print('Setting coordinates for (0, 0) top left corner')
+        self.tweak()
+        pos.extend([self.x.position, self.y.position])
+        print('Setting coordinates for (0, M) top right corner')
+        self.tweak()
+        pos.extend([self.x.position, self.y.position])
+        try:
+            # create presets
+            self.x.presets.add_hutch(value=pos[0], name="x_bottom_left")
+            self.x.presets.add_hutch(value=pos[2], name="x_top_left")
+            self.x.presets.add_hutch(value=pos[4], name="x_top_right")
+            self.y.presets.add_hutch(value=pos[1], name="y_bottom_left")
+            self.y.presets.add_hutch(value=pos[3], name="y_top_left")
+            self.y.presets.add_hutch(value=pos[5], name="y_top_right")
+        except AttributeError:
+            logger.warning('No folder setup for motor presets. '
```

Comment:
It'd be unfortunate to get to this point, after picking 3 points for it to say "sorry - it isn't setup correctly!"

Reply:
yes this would be very unfortunate indeed.... thank you

Reply:
Is there a neat way of checking if this is setup or do I just try to set it right after I asked for the first point and see if this works?

Reply:
`if not hasattr(self.y, 'presets')` or something, perhaps?

Reply:
Ok thanks